python/ray/_private/job_manager/job_manager.py

```diff
@@ -82,61 +87,60 @@ def get_status(self, job_id: str) -> JobStatus:
         assert pickled_status is not None, f"Status not found for {job_id}"
         return pickle.loads(pickled_status)
 
+    def put_child_process_pid(self, job_id: str, pid):
+        _internal_kv_put(
+            self.JOB_CHILD_PROCESS_PID_KEY.format(job_id=job_id),
```

Comment:
i don't think we should ever store PIDs like this. there is no way to know whether the pid has been recycled or something